archetype-common/src/main/java/org/apache/maven/archetype/creator/FilesetArchetypeCreator.java

```diff
@@ -702,7 +702,9 @@ private void rewriteEARPluginReferences( Plugin plugin, String rootArtifactId, S
         Xpp3Dom configuration = (Xpp3Dom) plugin.getConfiguration();
         if ( configuration != null )
         {
-            Xpp3Dom[] modules = configuration.getChild( "modules" ).getChildren();
+            Xpp3Dom[] modules = configuration.getChild( "modules" ) != null
```

Comment:
use a local variable for `configuration.getChild( "modules" )`.
That's better than this and better for debugging as well.

Reply:
Hi, I've incorporated your suggestion.